### ISSUE TITLE
sep commands db init broken

### ISSUE THREAD

**Issue by binary1230** [magfest/ubersystem#2980]
in our sep command we startup the DB like this:
``` Session.initialize_db(modify_tables=False, drop=False) ```

that used to work, but if you try and create any objects or modify stuff, you get the following error:
```qlalchemy.exc.UnboundExecutionError: ColumnDefault object is not bound to an Engine or Connection.  Execution can not proceed without a database to execute against.```

it appears if you replace the initialize_db line above with this line everything works again:
```
# normally called by on_startup() after cherrypy
# though that won't be called if you're in a sep command
initialize_db()
```

Is this a safe way to init the DB? anything weird with plugin ordering and registering their mixins before this happens?

If that's OK, then we should replace the stuff in sep_commands.py with it. maybe other places in the codebase

**Comment by binary1230** [comment]
to repro this, just do:
```
@entry_point
def whatever():
   Session.initialize_db(modify_tables=False, drop=False)
   Attendee()
```
then run ```sep whatever```

**Comment by RobRuana** [comment]
I think the correct way to do this is pass the `initialize=True` the way we do in `insert_admin()`:
```python
@entry_point
def insert_admin():
    Session.initialize_db(initialize=True)
    with Session() as session:
        if session.insert_test_admin_account():
            print("Test admin account created successfully")
        else:
            print("Not allowed to create admin account at this time")
```

I had to rework the way we initialize our database in order to get the `sep alembic` commands to work. I added the `initialize` parameter as a way of saying, "Ok, set up a connection to the database, but don't change anything!"

I think I just forgot to change the call to `Session.initialize_db()` in the rest of our `sep` commands 😬

**Comment by RobRuana** [comment]
Just double checked and this works like a charm:
```python
@entry_point
def whatever():
   Session.initialize_db(initialize=True)
   Attendee()
   print('All good!')
```

```
(env) vagrant@vagrant-ubuntu-trusty-64:~$ sep whatever
All good!
(env) vagrant@vagrant-ubuntu-trusty-64:~$ 
```

**Comment by kitsuta** [comment]
@binary1230 Are there any specific `sep` commands that are broken from this, or was this just something you found while creating new sep commands? Since this is just a lack of a flag, is it safe to close this issue?

**Comment by RobRuana** [comment]
This is not actually just a missing flag. `initialize_db()` and `Session.initialize_db()` are two entirely different functions. The one we should be calling is `Session.initialize_db()`, and we want to call it with `initialize=True`.

**Comment by RobRuana** [comment]
To clarify, the suggestion @binary1230 made in his initial comment is *NOT* correct, and is *NOT* what we should be doing:

> it appears if you replace the initialize_db line above with this line everything works again:
```
# normally called by on_startup() after cherrypy
# though that won't be called if you're in a sep command
initialize_db()
```

Instead of calling `initialize_db()` as originally suggested, we should be calling `Session.initialize_db(initialize=True)`.

**Comment by binary1230** [comment]
whatever Rob says is the right way to do it is correct.

**Comment by binary1230** [comment]
and commenting just that other 'sep' commands appear to initialize the DB in ways that don't allow write access, and we just never noticed til recently

**Comment by RobRuana** [comment]
Whoa, I've been poring over the entire codebase, and I came across the `initialize_db()` function you were referencing initially.

Your initial comment is correct, and the comment I made [here](https://github.com/magfest/ubersystem/issues/2980#issuecomment-352215637) is _NOT_ correct! For any sep commands that don't need to drop tables (`drop=True`), `initialize_db()` will do the correct thing, and is probably the preferred way to do it.

The two commands where `initialize_db()` won't work are `drop_uber_db` and `reset_uber_db`. I am guessing I was thinking of those commands when I wrote that initial comment.

In any case, ever since we switched to using `alembic` for migrations, we no longer need to do the extra work in `initialize_db()` to make sure that all the plugins have loaded their model mixins. We should probably remove the `initialize_db()` function entirely and just always use `Session.initialize_db()`.

Oh well! Just wanted to update you that you were right all along 😄
